### PR TITLE
tests: new tests.nested commands copy and wait-for

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This are the second part of the commands for the tests.nested tool
The new commands are:
	copy:        copies a file to the vm
	wait-for:    waits for an specified event (ssh. no-ssh,
snap-command and reboot)

Tests have been updated as well

### CODE REVIEW

File: tests/nested/core/core20-kernel-failover/task.yaml [hunk 1]
```diff
@@ -26,19 +26,10 @@ execute: |
...
-  wait=1
-  until tests.nested exec command -v snap; do
-      retry=$(( retry - 1 ))
-      if [ $retry -le 0 ]; then
-          echo "Timed out waiting for no snap command. Aborting!"
-          exit 1
-      fi
-      sleep "$wait"
-  done
+  tests.nested wait-for snap-command
```
Comment (mvo5) [inline tests/nested/core/core20-kernel-failover/task.yaml:32]: Thanks for this cleanup

**Review by mvo5 — APPROVED** [review]
Thank you

File: tests/lib/tools/tests.nested [hunk 2]
```diff
@@ -24,11 +26,19 @@ show_help() {
     echo "  remove-vm:   removes a vm"
     echo "  exec:        executes a command in the vm"
     echo "  exec-as:     executes a command in the vm as the specified user"
+    echo "  copy:        copies a file to the vm"
+    echo "  wait-for:    waits for an specified event"
```
Comment (stolowski) [inline tests/lib/tools/tests.nested:30]: s/an/a but probably the correct is:
```suggestion
    echo "  wait-for:    waits for the specified event"
```

File: tests/lib/tools/tests.nested [hunk 3]
```diff
@@ -24,11 +26,19 @@ show_help() {
...
     echo "IMAGE-TYPES:"
     echo "  core: work with a core image"
     echo "  classic: work with a classic image"
     echo ""
+    echo "EVENT:"
+    echo "  ssh: it is possible ssh to the vm"
+    echo "  no-ssh: is not possible to ssh to the vm"
+    echo "  snap-command: snap command is available in the vm"
+    echo "  reboot: reboot is done in the vm"
+    echo ""
```
Comment (stolowski) [inline tests/lib/tools/tests.nested:41]: Nice!

**Review by stolowski — APPROVED** [review]
Looks fine, I suggested a couple of small typo fixes.

File: tests/lib/tools/tests.nested (hunk 2 above)
Comment (sergiocazzolato) [reply tests/lib/tools/tests.nested:30]: updated

File: tests/lib/tools/tests.nested [hunk 4]
```diff
@@ -24,11 +26,19 @@ show_help() {
...
     echo "  exec-as:     executes a command in the vm as the specified user"
+    echo "  copy:        copies a file to the vm"
+    echo "  wait-for:    waits for an specified event"
     echo ""
     echo "IMAGE-TYPES:"
     echo "  core: work with a core image"
     echo "  classic: work with a classic image"
     echo ""
+    echo "EVENT:"
+    echo "  ssh: it is possible ssh to the vm"
```
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply tests/lib/tools/tests.nested:37]: updated

File: tests/nested/core/core20-kernel-failover/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/nested/core/core20-kernel-failover/task.yaml:32]: yaw

File: tests/lib/tools/tests.nested (hunk 3 above)
Comment (sergiocazzolato) [reply tests/lib/tools/tests.nested:41]: tx

**Pull request merged**